File: samples/Erlang/primality-by-trial-division.erl
is_prime(N) when N == 2 -> true;
is_prime(N) when N < 2 orelse N rem 2 == 0 -> false;
is_prime(N) -> is_prime(N,3).

is_prime(N,K) when K*K > N -> true;
is_prime(N,K) when N rem K == 0 -> false;
is_prime(N,K) -> is_prime(N,K+2).

File: samples/Go/primality-by-trial-division.go
func isPrime(p int) bool {
    if p < 2 { return false }
    if p == 2 { return true }
    if p % 2 == 0 { return false }

    for i := 3; i*i < p; i += 2 {
        if p % i == 0 { return false }
    }
    return true
}
